source/core_sntp_client.c

```diff
@@ -649,6 +649,11 @@ static SntpStatus_t processServerResponse( SntpContext_t * pContext,
                                            pContext->sntpPacketSize,
                                            &parsedResponse );
 
+        /* We do not expect the following errors to be returned as the context
+         * has been validated in the Sntp_ReceiveTimeResponse API. */
+        assert( status != SntpErrorBadParameter );
```

Comment:
This also introduces a potential DoS attack for unauthenticated connections (for example, if an attacker were to send a packet with a zero timestamp).

Not every usage of this library will have asserts disabled in production.

Reply:
The `SntpErrorBadParameter` status value is returned for data that is obtained from the client system (like request transmit and response receive time). An invalid response packet, if out of an attack, from the network results in other status codes.

Reply:
You're right. I mixed up SntpInvalidResponse and SntpErrorBadParameter.